src/Core/Managed/Shared/Extensibility/Implementation/TelemetryConfigurationFactory.cs

```diff
@@ -202,14 +204,48 @@ protected static void BuildTelemetryProcessorChain(XElement definition, Telemetr
                     builder = builder.Use(current =>
                     {
                         var constructorArgs = new object[] { current };
-                        return (ITelemetryProcessor)LoadInstance(addElement, typeof(ITelemetryProcessor), telemetryConfiguration, constructorArgs, null);
+                        return (ITelemetryProcessor)LoadInstance(addElement, typeof(ITelemetryProcessor), null, constructorArgs, null);
                     });
                 }
             }
 
             builder.Build();
         }
 
+        protected static void LoadTelemetrySinks(XElement definition, TelemetryConfiguration telemetryConfiguration)
```

Comment:
No, I am not trying to avoid `<Add>` element under `<Sinks>`, in fact, using `<Add>` is the way to create sinks. There is a couple of reasons why special handling for TelemetrySink is necessary

1. Just like for TelemetryProcessors, the sink has to be created from a constructor that takes parameters (telemetry configuration in the sink case)

2. As you can see, there is special logic involving default sink (always present, and designated by the name "default"). Default sink is never created.

3. Finally, if the sink is referred to multiple times in the configuration (by the same name), we apply all the configuration to the single, named instance. This part is debatable, but I had to choose some behavior in case I am facing repeated sink configuration with the same name and I think this is a reasonable choice.

Take a look at the tests and it will make things clearer.